### PR TITLE
Table: Implement Classes Option

### PR DESCRIPTION
**Pull request by cgack**
This includes the changes made in the table review branch https://github.com/jquery/jquery-mobile/tree/7360-table-review

### CODE REVIEW

**Comment by cgack** [comment]
TODOS
* [x] Implement Classes
* [ ] use hasClasses, lacksClasses in tests
* [ ] review QUnit.skip-ed tests

**Comment by cgack** [comment]
@arschmitz I think this is read for a round of review. Thanks!

**Comment by arschmitz** [comment]
on it

File: js/widgets/table.columntoggle.js [hunk 1]
```diff
@@ -20,239 +20,168 @@
...
-		// so it can be called on refresh, too
-
-		// update column toggles on resize
-		this._on( this.window, {
-			throttledresize: "_setToggleState"
-		} );
-		this._on( this._menu, {
-			"change input": "_menuInputChange"
-		} );
+	_updateVariableColumn: function( header, cells, priority/*, state */ ) {
```
Comment (arschmitz) [inline js/widgets/table.columntoggle.js]: remove unused stae var comment

File: js/widgets/table.columntoggle.js [hunk 2]
```diff
@@ -20,239 +20,168 @@
...
+			// after the refresh, are restored to their locked state
+			this._restoreLockedColumns( lockedColumns );
+		}
 	},
 
 	_destroy: function() {
-		this._super();
+		if ( this.options.mode === "columntoggle" ) {
+			if ( !this.options.enhanced ) {
+				this._removeClass( "ui-table-columntoggle" );
```
Comment (arschmitz) [inline js/widgets/table.columntoggle.js]: no need to remove classes in destroy

File: js/widgets/table.columntoggle.js [hunk 3]
```diff
@@ -20,239 +20,168 @@
...
+				this.headers.each( $.proxy( function( index, element ) {
+					var header,
+						priority = $.mobile.getAttribute( element, "priority" );
+
+					if ( priority ) {
+						header = $( element );
+						header
+							.add( header.jqmData( "cells" ) )
+								.removeClass( this.options.classes.priorityPrefix + priority );
```
Comment (arschmitz) [inline js/widgets/table.columntoggle.js]: same here

File: js/widgets/table.columntoggle.popup.js [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+return $.widget( "mobile.table", $.mobile.table, {
+	options: {
+		columnButton: true,
+		columnButtonTheme: null,
+		columnPopupTheme: null,
+		columnButtonText: "Columns...",
+		columnUi: true,
+		classes: {
+			"ui-table-columntoggle-popup": "",
+			"ui-table-columntoggle-btn": "ui-button ui-corner-all ui-shadow ui-mini"
```
Comment (arschmitz) [inline js/widgets/table.columntoggle.popup.js]: `ui-button` is not a theme class

File: js/widgets/table.js [hunk 5]
```diff
@@ -34,76 +34,102 @@ return $.widget( "mobile.table", {
...
-			} );
+		// Iterate over the header rows
+		this.headerRows.each( $.proxy( this, "_refreshHeaderRow" ) );
+	},
+
+	_destroy: function() {
+		var table = this.element;
+
+		if ( !this.options.enhanced ) {
+			this._removeClass( "ui-table" );
```
Comment (arschmitz) [inline js/widgets/table.js]: no need to remove classes in destory

File: js/widgets/table.reflow.js [hunk 6]
```diff
@@ -31,78 +31,95 @@
...
+			colstartAttr = "data-" + $.mobile.ns + "colstart";
+
+			if ( !this.options.enhanced ) {
+				this.element
+					.children( "thead" )
+						.find( "[" + colstartAttr + "]" )
+							.removeAttr( colstartAttr )
+						.end()
+					.end()
+					.removeClass( "ui-table-reflow" )
```
Comment (arschmitz) [inline js/widgets/table.reflow.js]: use of old removeclass and its in destory

**Comment by arschmitz** [comment]
A couple things i noticed throughout were use of jqmdata, jqmremovedata, and $.mobile.getAttribute all of which are deprecated. I also noticed there are not any backcompat tests over all looking good

**Comment by arschmitz** [comment]
Oh and of course no hasClasses yes but you have that as a todo already

**Comment by arschmitz** [comment]
Related to tests it looks like this fixes the failures we are seeing in 1.5-dev on table but the `Basic table: _refreshHeaderCell() ` test is failing across the board

**Comment by cgack** [comment]
gh-6531 needs to have the table-stripe removed

**Comment by arschmitz** [comment]
rebased fixed and landed on 1.5-dev

**Pull request closed without merging**